astroid/bases.py

```diff
@@ -461,19 +479,22 @@ def infer_call_result(
         # If we're unbound method __new__ of a builtin, the result is an
         # instance of the class given as first argument.
         if self._proxied.name == "__new__":
+            assert self._proxied.parent, "Expected a parent node"
             qname = self._proxied.parent.frame(future=True).qname()
             # Avoid checking builtins.type: _infer_type_new_call() does more validation
             if qname.startswith("builtins.") and qname != "builtins.type":
-                return self._infer_builtin_new(caller, context)
+                return self._infer_builtin_new(caller, context or InferenceContext())
         return self._proxied.infer_call_result(caller, context)
 
     def _infer_builtin_new(
         self,
-        caller: nodes.Call,
+        caller: SuccessfulInferenceResult | None,
         context: InferenceContext,
     ) -> collections.abc.Generator[
         nodes.Const | Instance | UninferableBase, None, None
     ]:
+        if not isinstance(caller, nodes.Call):
```

Comment:
As expected. Again, I think it makes more sense to bring this in line with what we see in the wild and build robust code (without tests) for it, than spending a lot of time to see why we are passing arguments to this function that are too broad.